### PR TITLE
NXBT-3761: Move API Plaground to Platform CI for Nuxeo 2023

### PR DESCRIPTION
**Pull request by kevinleturc**

### CODE REVIEW

**Comment by kevinleturc** [comment]
Successful build: https://jenkins.platform-staging.dev.nuxeo.com/blue/organizations/jenkins/api-playground%2Fnuxeo-api-playground/detail/PR-10/6/pipeline/

File: ci/Jenkinsfiles/build.groovy [hunk 1]
```diff
@@ -0,0 +1,202 @@
...
+  container('maven') {
+    def nuxeoParentVersion = readMavenPom().getParent().getVersion()
+    // target connect preprod if nuxeo-parent is a snapshot version or a build version
+    return nuxeoParentVersion.matches("^\\d+\\.\\d+(-SNAPSHOT|\\.\\d+)\$") ? 'instance-clid-preprod' : 'instance-clid'
+  }
+}
+
+pipeline {
+  agent {
+    label 'jenkins-nuxeo-jsf-lts-2023'
```
Comment (ataillefer) [inline ci/Jenkinsfiles/build.groovy]: The package pod template is not enough?

File: ci/docker/skaffold.yaml [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+          SCM_REF: "{{.GIT_COMMIT}}"
+          VERSION: "{{.VERSION}}"
+  cluster:
+    namespace: platform
+    dockerConfig:
+      secretName: jenkins-docker-cfg
+    resources:
+      requests:
+        cpu: "1"
+        memory: "3Gi"
```
Comment (ataillefer) [inline ci/docker/skaffold.yaml]: Should we try having request=limit?

File: ci/Jenkinsfiles/build.groovy (hunk 1 above)
Comment (kevinleturc) [reply ci/Jenkinsfiles/build.groovy]: No, we need to have dart which is only installed in the `builder-java\d+-firefox42-nodejs\d+` which is not part of package pod template, which uses `builder-java\d+-chrome-nodejs\d+`, but it is used by the jsf pod template.

File: ci/docker/skaffold.yaml (hunk 2 above)
Comment (kevinleturc) [reply ci/docker/skaffold.yaml]: Yes we could give a first try on this.
I'll set:
```yaml
    resources:
      requests:
        cpu: "1"
        memory: "4Gi"
      limits:
        memory: "4Gi"
```
Which meet the best practices, Memory request == limit, and no CPU limit. Furthermore these settings follows what we have on the underlying Kubernetes nodes, ie: Memory = 4 x CPU.

**Review by troger — DISMISSED** [review]

**Comment by kevinleturc** [comment]
The branch `2023` was created with a minimal build.groovy in order to have the commit of this PR reported to Jira.

The PR can now be merged to `2023` branch.

**Review by ataillefer — APPROVED** [review]

**Pull request merged**